src/main/java/au/com/agiledigital/idea_search/dao/FedexIdeaDao.java

```diff
@@ -0,0 +1,281 @@
+                this.ao.find(
+                        AO_FEDEX_IDEA_TYPE,
+                        Query.select().where("CONTENT_ID = ?", contentId)
+                );
+        log.error(String.valueOf(aoFedexIdeas.length));
+        // If the title of the page already exists the create event fails and a page update event is fired on a successful save.
+        // If this happens the aoFedexIdea will not exist in the data store and will need to be created.
+        AoFedexIdea aoFedexIdea = aoFedexIdeas.length == 0 ? this.ao.create(AO_FEDEX_IDEA_TYPE) : aoFedexIdeas[0];
+
+
+        List<AoFedexTechnology> aoFedexTechnology = Arrays.asList(
+                aoFedexIdea.getTechnology()
+        );
+
+        // Deletes the technology item from the table
+        aoFedexTechnology.forEach(this.ao::delete);
+
+        List<AoFedexTechnology> aoTechList = getAoFedexTechnologies(fedexIdea);
+
+        this.prepareAOFedexIdea(aoFedexIdea, fedexIdea);
+        SetTechnology(aoTechList, aoFedexIdea);
+        aoFedexIdea.save();
+
+        return this.asFedexIdea(aoFedexIdea);
+    }
+
+    /**
+     * Extract technologies from FedexIdea and create an active object for each
+     *
+     * @param fedexIdea from model
+     * @return List<AoFedexTechnology> unsaved.
+     */
+    @NotNull
+    private List<AoFedexTechnology> getAoFedexTechnologies(FedexIdea fedexIdea) {
+        return fedexIdea
+                .getTechnologies()
+                .stream()
+                .map(tech -> this.prepareAOFedexTechnology(tech.getTechnology()))
+                .collect(Collectors.toList());
+    }
+
+    //    See comment on https://community.atlassian.com/t5/Jira-questions/ActiveObjects-jira/qaq-p/354375
+    //    This means that the setter is done on the Recipient entity(setting the Filter) and because of the way relational databases...
+    private void SetTechnology(
+            List<AoFedexTechnology> aoFedexTechnologies,
+            AoFedexIdea aoFedexIdea
+    ) {
+        aoFedexTechnologies.forEach(
+                techItem -> {
+                    techItem.setIdea(aoFedexIdea);
+                    techItem.save();
+                }
+        );
+    }
+
+    /**
+     * List all fedex idea in the data store
+     *
+     * @return List<FedexIdea>
+     */
+    public List<FedexIdea> findAll() {
+        AoFedexIdea[] aoFedexIdeas = this.ao.find(AO_FEDEX_IDEA_TYPE);
+        return this.asListFedexIdea(aoFedexIdeas);
+    }
+
+    /**
+     * List all fedex idea technology in the data store
+     *
+     * @return List<FedexTechnology>
+     */
+    public List<FedexTechnology> findAllTech() {
+        AoFedexTechnology[] aoFedexTechnologies =
+                this.ao.find(AO_FEDEX_TECHNOLOGY_TYPE, Query.select());
+        return this.asListFedexTechnology(aoFedexTechnologies);
+    }
+
+    /**
+     * Convert array of active objects to a list of model objects
+     *
+     * @param aoFedexIdeas AoFedexIdea[]
```

Comment:
Should this be
```suggestion
     * @param aoFedexIdeas Some description here?
```
As the type information is inferred?

Reply:
added